pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -128,10 +128,9 @@ func (cb *ClusterBuilder) buildSubsetCluster(opts buildClusterOpts, destRule *co
 		clusterType = cluster.Cluster_ORIGINAL_DST
 	}
 	if !(isPassthrough || clusterType == cluster.Cluster_EDS) {
-		if len(subset.Labels) != 0 {
-			lbEndpoints = cb.buildLocalityLbEndpoints(proxyNetworkView, service, opts.port.Port, []labels.Instance{subset.Labels})
-		} else {
-			lbEndpoints = cb.buildLocalityLbEndpoints(proxyNetworkView, service, opts.port.Port, nil)
+		lbEndpoints = cb.buildLocalityLbEndpoints(proxyNetworkView, service, opts.port.Port, []labels.Instance{subset.Labels})
+		if len(lbEndpoints) == 0 {
+			log.Warnf("locality endpoints missing for cluster %s", subsetClusterName)
```

Comment:
When does this happen and what should a user do if they see this warning?

Reply:
This happens if subset clusters with DNS resolution does not have members - happened for us in Service Entry (primarily because of this regression #33865  - it took quite some time to figure out that is the problem.

Reply:
Hmm, my only concern is that its not actually a misconfig/bug to have zero, so there may be quite a bit of spam?

Reply:
>  I am imaging you see 503 UH, then can look at the cluster from config dump and see no endpoints and figure out something wrong.

Yes. We are seeing 503s and we are getting "no_cluster" incremented on the route.  This cluster is not available in config_dump - I suspect because Envoy may be rejecting it as it is a STRICT_DNS cluster with no endpoints - I could not find that in logs. I could not find why the cluster was not created. That was the motivation. I agree it is not a misconfiguration but a bug in Istio. If Envoy NACKs are available this is is not needed though.